### PR TITLE
implement attack

### PR DESCRIPTION
**Pull request by lawand7**

### CODE REVIEW

File: src/calculationEngine/entities/Attack.java [hunk 1]
```diff
@@ -1,10 +1,49 @@
...
+    private int calculateRandomLevel() {
+        Random random = new Random();
+        return props[random.nextInt(props.length)];
+    }
+
+
+    public void levelUp() {
+        if (level < maxLevel) {
+            level++;
+        }
```
Comment (lawand7) [inline src/calculationEngine/entities/Attack.java:47]: not sure about this to be honest. Let's just discuss it, if needed

Comment (BigKahuna7385) [reply src/calculationEngine/entities/Attack.java:47]: I think you're missing a function, which scales all values according to the level of the attack.
E. g. in levelUp you could call it to raise all necessary stats by a certain amounts.
And in the constructor you can call it, to scale everything according to the level of the attack.

**Review by BigKahuna7385 — CHANGES_REQUESTED** [review]

File: src/calculationEngine/entities/Attack.java [hunk 2]
```diff
@@ -1,10 +1,102 @@
...
 
-    // Placeholder for Attack logic
+    public Attack(String name, int damage, int accuracy, int criticalChance, BeastTypes type, int executions, int level) {
+        this.name = name;
+        this.damage = damage;
+        this.accuracy = accuracy;
+        this.criticalChance = criticalChance;
+        this.type = type;
+        this.level = level;
+        this.executions = executions;
```
Comment (BigKahuna7385) [inline src/calculationEngine/entities/Attack.java:35]: scalestats() also needs to be in here.

File: src/calculationEngine/entities/Attack.java [hunk 3]
```diff
@@ -1,10 +1,102 @@
...
+        scaleStats();
+    }
+
+    private void scaleStats() {
+        //e.g. level3 increases stats by 30%
+        double factor = 1 + level * 0.1;
+        damage = (int) Math.round(damage * factor);
+        accuracy = (int) Math.round(accuracy * factor);
+        criticalChance = (int) Math.round(criticalChance * factor);
+    }
```
Comment (BigKahuna7385) [inline src/calculationEngine/entities/Attack.java:57]: this. to get the correct scope. (not really needed but I like it more)

**Review by BigKahuna7385 — CHANGES_REQUESTED** [review]

**Review by BigKahuna7385 — APPROVED** [review]

**Pull request merged**